src/injector.js

```diff
@@ -57,8 +57,10 @@ export default class Injector {
       flatten = list => list.reduce(
         (a, b) => a.concat(Array.isArray(b) ? flatten(b) : b), []
       ),
-      injectIntoDOM = (dependencies, idx = 0) => {
-        const elem = dependencies[idx];
+      injectIntoDOM = (dependencies, idx = 0, type = 'printed') => {
+        if (idx >= dependencies.length) { return; }
+
+        const elem = dependencies[idx][type];
 
         if (elem === undefined) { return; }
         else if (elem.getAttribute('data-dactylographsy-uncached-js')) {
```

Comment:
Unsure about this. Wondering if we shouldn't switch over the `type` instead but since the dependencies are a flat list here we lost the information here.